react-components/table-intro/step8.md

```diff
@@ -1,118 +1,64 @@
-In this step, we'll add in the heart of pagination functionality. We'll add a new state property to track the dynamic rows, and a few more handler functions to facilitate updating the various state properties based off of user interactions.
+In this next step, we'll add the state properties we need to track how many rows per page the user can display, what current page the user is on within the dataset, and we'll add a function to handle updating the number of rows per page to display when the user changes this value with the dropdown.
 
-## Task
+1) <strong>Create a state property to track how many rows to display per page.</strong>
 
-1) Create a new state property to represent our dynamic rows, call it "rows". Set the initial value to the defaultRows constant that you defined earlier. Place this code just above the handlePerPageSelect function you added in the last step.
+In the `src/App.js` file, just below the line that starts with `const App = () => {`, add the following code to the App component. This code creates a state variable called `numPerPage` and sets the default value to `2`, so that we will see 2 rows per page by default.
 
 <pre class="file">
-const [rows, setRows] = React.useState(defaultRows);
+const defaultPerPage = 2;
+const [numPerPage, setNumPerPage] = React.useState(defaultPerPage);
 </pre>
 
-2) Locate the code that sets the rows for the table (`rows={defaultRows}`), and update it to reference the new rows state property you just created.
+<strong>Note:</strong> The second parameter in the destructured array returned by React.useState() is the setter for the state variable you're creating. These "setters" are functions that update the corresponding state value.
+
+2) <strong>Set the `perPage` prop of the `<Pagination />` component to the `numPerPage` state property you just created.</strong>
 
 <pre class="file">
-rows={rows}
+perPage={numPerPage}
 </pre>
 
-<strong>Note: </strong> the itemCount prop will still reference the defaultRows constant as itemCount needs to be aware of the total number of rows in the entire dataset, not the number of rows that's being shown at a given time.
+3) <strong>Create a state property to track the current page of the table data.</strong>
 
-3) Add a function "updateList" just above handlePerPageSelect to handle setting the rows based on when the numPerPage state property changes.
+Add the following code to the App component, just below the previous state property you created. Supply a `1` as the default value so that the table intuitively loads with the first page worth of data.
```

Comment:
```suggestion
Add the following code to the App component, just below the previous state property that was added in step []. Supply a `1` as the default value so that the table intuitively loads with the first page of data.
```